### ISSUE TITLE
System.NotImplementedException : 'Model load of h5 format has not been supported

### ISSUE THREAD

**Issue by Cilouche** [SciSharp/TensorFlow.NET#1024]
Hello,
I am trying to load a .h5 model with the load_model function (Tensorflow.Keras.Models) but I get this error
issue: 
System.NotImplementedException : 'Model load of h5 format has not been supported. Please submit an issue to https://github.com/SciSharp/TensorFlow.NET/issues if it's needed.'

any suggestions please,
Thank you

**Comment by AsakusaRinne** [comment]
Hello, the loading of "h5" format keras model has not been supported. We has implemented the loading of "SavedModel" format first because it's recommended by tensorflow. The loading of "h5" format is also on our schedule but is not very urgent now. May I ask why you choose "h5" format instead of "SavedModel"?

**Comment by Cilouche** [comment]
Thank you for your answer, I chose the .h because it is a network that I have already trained in Python with tensorflow-keras, and since I work in c# I wanted to reuse it

**Comment by AsakusaRinne** [comment]
I see. The loading of "h5" format is a big feature and support of the loading complex models (such as bert) with "SavedModel" format has higher priority in tf.net. We will implement the loading of of "h5" format model in the future but it costs time. A possible alternative is loading your model in python and save it again to "SavedModel" format, then load it in C#. If your model contains self-defined keras module, please try after the merge of #1022, which added support for it.

**Comment by Cilouche** [comment]
in python it loads normal (.h and .pb format) but in c# no

import tensorflow as tf

#import tensorflow.python.framework.convert_variables_to_constants_v2 
from tensorflow.python.keras.models import load_model
from tensorflow.keras.models import Model
model =  tf.keras.models.load_model('mod/InceptionResNet/')
print(model.summary())

**Comment by AsakusaRinne** [comment]
Hi, the file seems to be SavedModel format saved prior to TF 2.5. The loading itself is supported by using `tf.saved_model.load`. However it seems that your model has other parts such as restoring optimizer so it will still fail. I'll finish the lacked features in the next 2 or 3 days.

**Comment by Cilouche** [comment]
Yes

![image](https://user-images.githubusercontent.com/48053323/233405376-e1977b12-018c-4ba6-8f4c-262fe371b779.png)

**Comment by AsakusaRinne** [comment]
> Yes
> 
> ![image](https://user-images.githubusercontent.com/48053323/233405376-e1977b12-018c-4ba6-8f4c-262fe371b779.png)

Thanks for your sharing. That's a very good work around. I'll still fix the bug of loading your uploaded model in this issue.

**Comment by Cilouche** [comment]
No, I didn't delete anything, as indicated above, I opted for a format.h5 conversion to frozen model.pb with a well-detailed tutorial.

**Comment by AsakusaRinne** [comment]
Did you succeed in tensorflow (python)? I tried loading the file with tensorflow python but failed, which tells me it cannot find variable files.

**Comment by Cilouche** [comment]
a problem with this model that I put in attachment, I use another model, basically I took my basic model .h5 convert to .pb, but the attached model is wrong

**Comment by AsakusaRinne** [comment]
Thanks, I'll close this issue as completed. Please don't hesitate to tell us if any problem in the future.

**Comment by jaroslav-hook** [comment]
how about the loading a model?  You seem to have implemented model.save , model.save_weights   which saves the TF model.
How about loading it ? Cannot find it. 
By the way , where  why is               model.predict() requires a different  data format from model.fit()?
